### PR TITLE
Pending enterprise customer admin endpoint

### PR DESCRIPTION
**Pull request by MueezKhan246**
Description:
Added endpoint for pending_enterprise_customer_admin_user.

JIRA:
https://2u-internal.atlassian.net/jira/software/c/projects/ENT/issues/ENT-8797

### CODE REVIEW

File: enterprise/api/v1/views/pending_enterprise_customer_admin_user.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+from enterprise import models
+from enterprise.api.v1 import serializers
+from enterprise.api.v1.views.base_views import EnterpriseReadWriteModelViewSet
+from enterprise.logging import getEnterpriseLogger
+
+LOGGER = getEnterpriseLogger(__name__)
+
+
+class PendingEnterpriseCustomerAdminUserViewSet(EnterpriseReadWriteModelViewSet):
```
Comment (sameenfatima78) [inline enterprise/api/v1/views/pending_enterprise_customer_admin_user.py:16]: Question: How do we ensure that a user who is already a part of a customer does not get added to the PendingEnterpriseCustomerAdmin list?

File: tests/test_enterprise/api/test_views.py [hunk 2]
```diff
@@ -838,6 +839,77 @@ def test_post_pending_enterprise_customer_user_logged_out(self):
...
+    def setup_admin_user(self, is_staff=True):
+        """
+        Creates an admin user and logs them in
+        """
+        client_username = 'client_username'
+        self.client.logout()
+        self.create_user(username=client_username, password=TEST_PASSWORD, is_staff=is_staff)
+        self.client.login(username=client_username, password=TEST_PASSWORD)
+
+    def test_post_pending_enterprise_customer_admin_user_creation(self):
```
[the remark this answers is not in the document]
Comment (MueezKhan246) [reply tests/test_enterprise/api/test_views.py:872]: updated

**Review by sameeramin — APPROVED** [review]
LGTM 👍🏼

**Review by zamanafzal — APPROVED** [review]

**Review by hamzawaleed01 — APPROVED** [review]
Good work @MueezKhan246 👏

**Review by sameenfatima78 — COMMENTED** [review]
Added some comments. We can create a follow up ticket to address these.

File: enterprise/api/v1/serializers.py [hunk 3]
```diff
@@ -1624,6 +1625,51 @@ class Meta:
...
+                enterprise_customer=enterprise_customer
+            ).exclude(id=instance.id)
+        else:
+            existing_instances = PendingEnterpriseCustomerAdminUser.objects.filter(
+                user_email=user_email,
+                enterprise_customer=enterprise_customer
+            )
+
+        if existing_instances.exists():
+            raise serializers.ValidationError('A pending user with this email and enterprise customer already exists.')
```
[the remark this answers is not in the document]
Comment (MueezKhan246) [reply enterprise/api/v1/serializers.py:1659]: @sameenfatima78 i checked the constraint through test case to differentiate between the responses,
for some reason the constraint was not being enforced. i might have to check again.

**Pull request merged**